cli/src/merge_tools/external.rs

```diff
@@ -170,8 +170,17 @@ pub enum ExternalToolError {
          --debug to see the exact invocation)"
     )]
     InvalidConflictMarkers { exit_status: ExitStatus },
+    #[error(
+        "The output file is either unchanged or empty after the editor quit (run with --debug to \
+         see the exact invocation)."
+    )]
+    EmptyOrUnchanged,
+    #[error(transparent)]
+    InvalidRepoPath(#[from] InvalidRepoPathError),
+    #[error(transparent)]
+    Backend(#[from] jj_lib::backend::BackendError),
     #[error("I/O error")]
-    Io(#[source] std::io::Error),
+    Io(#[from] std::io::Error),
```

Comment:
This is source of bugs because there are other `io::Error` variants. I'm not pretty sure what this `Io` error represents, but it's not the error of console I/O.